libyul/backends/evm/EVMDialect.cpp

```diff
@@ -161,11 +162,12 @@ map<YulString, BuiltinFunctionForEVM> createBuiltins(langutil::EVMVersion _evmVe
 				_assembly.appendConstant(0);
 			else
 			{
+				auto [isData, subIdPath] = _context.currentObject->pathToSubObject(dataName.str());
 				yulAssert(
-					_context.subIDs.count(dataName) != 0,
+					!subIdPath.empty() | isData,
```

Comment:
```suggestion
					!subIdPath.empty() || isData,
```
Also, this is a completely different assertion. I think you still need the old one in addition to this when you're dealing with `Data`:
```c++
yulAssert(!isData || _context.subIDs.count(dataName) != 0, "");
```